test/hotspot/jtreg/compiler/lib/ir_framework/IRNode.java

```diff
@@ -1747,6 +2218,151 @@ public static boolean isCompositeIRNode(String irCompositeNodeString) {
         return irCompositeNodeString.startsWith(COMPOSITE_PREFIX);
     }
 
+    /**
+     * Is {@code irVectorNodeString} an IR vector node placeholder string?
+     */
+    public static boolean isVectorIRNode(String irVectorNodeString) {
+        return irVectorNodeString.startsWith(VECTOR_PREFIX);
+    }
+
+    /**
+     * Is {@code irVectorSizeString} a vector size string?
+     */
+    public static boolean isVectorSize(String irVectorSizeString) {
+        return irVectorSizeString.startsWith(VECTOR_SIZE);
+    }
+
+    /**
+     * Parse {@code sizeString} and generate a regex pattern to match for the size in the IR dump.
+     */
+    public static String parseVectorNodeSize(String sizeString, String typeString, VMInfo vmInfo) {
+        if (sizeString.equals("any")) {
+            return "\\\\d+"; // match with any number
+        }
+        // Try to parse any tags, convert to comma separated list of ints
+        sizeString = parseVectorNodeSizeTag(sizeString, typeString, vmInfo);
+        // Parse comma separated list of numbers
+        String[] sizes = sizeString.split(",");
+        String regex = "";
+        for (int i = 0; i < sizes.length; i++) {
+            int s = 0;
+            try {
+                s = Integer.parseInt(sizes[i]);
```

Comment:
I'm reporting an error on negative numbers now. But I'm not sure I want to limit it to power of 2 or a size limit. Especially when uses flags/tags this could not always hold.